src/search-field.jsx

```diff
@@ -0,0 +1,260 @@
+const React = require('react');
+const ReactTransitionGroup = require('react-addons-transition-group');
+const StylePropable = require('./mixins/style-propable');
+const ClickAwayable = require('./mixins/click-awayable');
+const TextField = require('./text-field');
+const Menu = require('./menus/menu');
+const MenuItem = require('./menus/menu-item');
+
+const SearchField = React.createClass({
+
+  mixins: [
+    StylePropable,
+    ClickAwayable,
+  ],
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  propTypes: {
+    errorText: React.PropTypes.string,
+    floatingLabelText: React.PropTypes.string,
+    errorStyle: React.PropTypes.object,
+    hintText: React.PropTypes.string,
+    id: React.PropTypes.string,
+    onChange: React.PropTypes.func,
+    autoWidth: React.PropTypes.bool,
+    menuStyle: React.PropTypes.object,
+    listStyle: React.PropTypes.object,
+    menuProps: React.PropTypes.object,
+    menuCloseDelay: React.PropTypes.number,
+    onUpdateRequests: React.PropTypes.func,
+    onNewSearchRequest: React.PropTypes.func,
+  },
+
+  getDefaultProps() {
+    return {
+      fullWidth: false,
+      open: false,
+      searchText: '',
+      menuCloseDelay: 100,
+
+      onChange: () => {},
+      onUpdateRequests: () => {},
```

Comment:
I'm wondering what his the best way to give the component matches he should display. You are currently using a synchrone property called `onUpdateRequests`.
I that it's better to use an object property called something like `source` or whatever.